app.py

```diff
@@ -116,62 +147,105 @@ def process_tweet(tweet, twapi):
             sig = tweet_parts[-1]
             pub_key = tweet_parts[-2]
             sig_valid = is_sig_valid(sig, screen_name, pub_key)
+            log_line += (
+                f', sig="{sig}", pub_key="{pub_key}"'
+                f', sig_valid="{sig_valid}"'
+            )
             if sig_valid:
                 store_verified_party(screen_name, pub_key)
                 store_tweet_record(tweet_id, screen_name, "PASSED")
-                msg = f"@{screen_name} {success_reply}"
+                msg = f"@{screen_name} {TWITTER_REPLY_SUCCESS}"
+                log_line += f', reply_msg="{msg}"'
                 twapi.update_status(status=msg, in_reply_to_status_id=tweet_id)
-                print("signature verified!")
+                replied_to_user = True
+                log_line += ', status="SUCCESS"'
             else:
-                print("invalid sig")
                 store_tweet_record(tweet_id, screen_name, "INVALID")
-                msg = f"@{screen_name} {signature_invalid_reply}"
+                msg = f"@{screen_name} {TWITTER_REPLY_INVALID_SIGNATURE}"
+                log_line += f', reply_msg="{msg}"'
                 twapi.update_status(status=msg, in_reply_to_status_id=tweet_id)
+                replied_to_user = True
+                log_line += ', status="INVALID"'
         else:
-            print("tweet already processed")
-    except Exception:
+            log_line += ', status="SKIP"'
+            replied_to_user = False
+    except Exception as err:
         traceback.print_exc()
-        print("cannot parse tweet")
-        msg = f"@{screen_name} {format_invalid_reply}"
+        print(err)
+        msg = f"@{screen_name} {TWITTER_REPLY_INVALID_FORMAT}"
+        log_line += f', error="{err}", reply_msg="{msg}"'
         twapi.update_status(status=msg, in_reply_to_status_id=tweet_id)
+        replied_to_user = True
         store_tweet_record(tweet_id, screen_name, "UNPARSEABLE")
+        log_line += ', status="ERROR"'
+    finally:
+        end_time = timer()
+        elapsed_time_ms = int((end_time - start_time) * 1000)
+        log_line += f', time_ms="{elapsed_time_ms}"'
+        print(log_line)
+
+    return replied_to_user
 
 
-def search_tweets():
-    twapi = Twython(consumer_key, consumer_secret, access_token, access_secret)
+def search_tweets(twapi: Twython) -> list:
+    start_time = timer()
+    log_line = "Searching tweets"
     try:
         results = twapi.search(
-            q=twitter_search_text, count=50, tweet_mode="extended"
+            q=TWITTER_SEARCH_TEXT, count=50, tweet_mode="extended"
```

Comment:
We should check what the maximum value is that can be used for `count`, maybe we can externalise this with an environment variable as well so that it can be changed?